internal/zinc-compile-core/src/main/scala/sbt/internal/inc/ProblemStringFormats.scala

```diff
@@ -0,0 +1,32 @@
+package sbt
+package internal
+package inc
+
+import xsbti.Problem
+import sbt.util.ShowLines
+import sbt.util.InterfaceUtil.jo2o
+
+/**
+ * The string representation of compiler warnings and error messages,
+ * used by LoggerReporter and the logging system.
+ */
+trait ProblemStringFormats {
```

Comment:
Why make it a trait and then create an anonymous class to use it as `problemStringFormats`? Wouldn't it be better to make it an object?